pyOCD/target/cortex_m.py

```diff
@@ -488,7 +488,10 @@ def writeMemory(self, addr, value, transfer_size = 32):
         write a memory location.
         By default the transfer size is a word
         """
-        self.transport.writeMem(addr, value, transfer_size)
+        try:
+            self.transport.writeMem(addr, value, transfer_size)
+        except:
```

Comment:
Perhaps. Confirming that a TAP reset occurred vs some other error would take further inspection. The DP CTRL/STAT register would need to be inspected. Where do you think it makes sense to do this?